app/main.py

```diff
@@ -1,2 +1,54 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.length = 0
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.threshold = self.capacity * self.load_factor
+        self.hash_table = [[]for _ in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+        if self.length > self.threshold:
+            self.rebuilding_hash_table()
+        self.add_new_element(key, value)
+
+    def __getitem__(self, item):
+        hashed_value = hash(item)
+        index = hashed_value % self.capacity
+        while self.hash_table[index]:
+            if self.hash_table[index][1] == item \
+                    and self.hash_table[index][0] == hashed_value:
+                return self.hash_table[index][2]
+            index = (index + 1) % self.capacity
+        raise KeyError
+
+    def rebuilding_hash_table(self):
+        self.capacity *= 2
+        self.threshold = self.capacity * self.load_factor
+        prev_list = [item for item in self.hash_table if item]
+        self.hash_table = [[] for _ in range(self.capacity)]
+        for item in prev_list:
+            index = item[0] % self.capacity
+            while True:
+                if len(self.hash_table[index]) > 0:
+                    index = (index + 1) % self.capacity
+                else:
+                    self.hash_table[index] = item
+                    break
```

Comment:
use setitem here

Reply:
don't understand

Reply:
create a loop, and use the magic method` __setitem__`

Reply:
for item in prev_list:
            self.__setitem__(item[1], item[2])

like this?

Reply:
yes

Reply:
if i do like this, program don`t work

Reply:
in __setitem__ i call method rebuilding_hash_table

Reply:
Maybe you mean i need call method "add_new_element"?

Reply:
<img width="808" alt="image" src="https://user-images.githubusercontent.com/72562232/186160531-b577215c-7987-4950-ab31-dcdbe98b36a7.png">
it should work correctly if you add self.length = 0.